web/templates/move_topic_to_stream.hbs

```diff
@@ -1,26 +1,50 @@
 {{#unless (or from_message_actions_popover only_topic_edit)}}
-<p class="white-space-preserve-wrap">{{#tr}}Move all messages in <strong>{topic_name}</strong>{{/tr}} to:</p>
+<p>
+    {{#tr}}
+    You are moving all messages from <z-stream></z-stream>
+    {{#*inline "z-stream"}}<b>{{> inline_decorated_stream_name stream=stream show_colored_icon=true}}</b>{{/inline}}
+    {{/tr}}
+    <i class="fa fa-angle-right" aria-hidden="true"></i>
+    {{#tr}}<b>{topic_name}.</b>{{/tr}}
+</p>
 {{/unless}}
 <form id="move_topic_form">
     {{#if only_topic_edit }}
     <p>{{t "Rename topic to:" }}</p>
     {{else if from_message_actions_popover}}
-    <p>{{t "Move messages to:" }}</p>
+    <p>
+        {{#tr}}
+        You are moving messages from <z-stream></z-stream>
+        {{#*inline "z-stream"}}<b>{{> inline_decorated_stream_name stream=stream show_colored_icon=true}}</b>{{/inline}}
+        {{/tr}}
+        <i class="fa fa-angle-right" aria-hidden="true"></i>
+        {{#tr}}<b>{topic_name}.</b>{{/tr}}
+    </p>
     {{/if}}
     <div class="topic_stream_edit_header">
         {{#unless only_topic_edit}}
-        {{> dropdown_widget_wrapper widget_name="move_topic_to_stream"}}
-        <i class="fa fa-angle-right" aria-hidden="true"></i>
+        <div class="input-group">
+            <label class="modal-field-label">New channel</label>
+            {{> dropdown_widget_wrapper widget_name="move_topic_to_stream"}}
+        </div>
         {{/unless}}
-        <input name="new_topic_name" type="text" class="move_messages_edit_topic modal_text_input" autocomplete="off" value="{{topic_name}}" {{#if disable_topic_input}}disabled{{/if}} />
+        <div class="input-group">
+            {{#unless only_topic_edit}}
+            <label for="new_topic" class="modal-field-label">New topic</label>
```

Comment:
`for` should never point to something that doesn't exist.

Reply:
Probably you want to add an `move-topic-new-topic-name` ID on the input and then point this `for` at that.